sync/src/testFixtures/java/tech/pegasys/teku/sync/multipeer/batches/StubBatch.java

```diff
@@ -0,0 +1,193 @@
+  private final List<SignedBeaconBlock> blocks = new ArrayList<>();
+  private boolean invalid = false;
+  private Optional<Runnable> blockCallback = Optional.empty();
+  private boolean complete = false;
+  private boolean contested = false;
+  private boolean firstBlockConfirmed = false;
+  private boolean lastBlockConfirmed = false;
+
+  public StubBatch(final TargetChain targetChain, final UInt64 firstSlot, final UInt64 count) {
+    checkArgument(
+        count.isGreaterThanOrEqualTo(UInt64.ONE), "Must include at least one slot in a batch");
+    this.targetChain = targetChain;
+    this.firstSlot = firstSlot;
+    this.count = count;
+  }
+
+  public void receiveBlocksAndMarkComplete(final SignedBeaconBlock... blocks) {
+    receiveBlocks(blocks, true);
+  }
+
+  public void receiveBlocks(final SignedBeaconBlock... blocks) {
+    // Batches automatically mark themselves complete when the block is received for the last slot
+    // or a request comes back completely empty
+    final boolean markComplete =
+        blocks.length == 0 || blocks[blocks.length - 1].getSlot().equals(getLastSlot());
+    receiveBlocks(blocks, markComplete);
+  }
+
+  public void requestError() {
+    blockCallback.orElseThrow().run();
+    blockCallback = Optional.empty();
+  }
+
+  private void receiveBlocks(final SignedBeaconBlock[] blocks, final boolean markComplete) {
+    checkState(blockCallback.isPresent(), "Received blocks when none were expected");
+    this.blocks.addAll(asList(blocks));
+    if (markComplete) {
+      complete = true;
+    }
+    final Runnable callback = blockCallback.get();
+    blockCallback = Optional.empty();
+    callback.run();
+  }
+
+  @Override
+  public UInt64 getFirstSlot() {
+    return firstSlot;
+  }
+
+  @Override
+  public UInt64 getLastSlot() {
+    return firstSlot.plus(count).minus(1);
+  }
+
+  @Override
+  public Optional<SignedBeaconBlock> getFirstBlock() {
+    return blocks.isEmpty() ? Optional.empty() : Optional.of(blocks.get(0));
+  }
+
+  @Override
+  public Optional<SignedBeaconBlock> getLastBlock() {
+    return blocks.isEmpty() ? Optional.empty() : Optional.of(blocks.get(blocks.size() - 1));
+  }
+
+  @Override
+  public List<SignedBeaconBlock> getBlocks() {
+    return blocks;
+  }
+
+  @Override
+  public void markComplete() {
+    complete = true;
+  }
+
+  @Override
+  public boolean isComplete() {
+    return complete;
+  }
+
+  @Override
+  public boolean isConfirmed() {
+    return firstBlockConfirmed && lastBlockConfirmed;
+  }
+
+  @Override
+  public boolean isFirstBlockConfirmed() {
+    return firstBlockConfirmed;
+  }
+
+  @Override
+  public boolean isContested() {
+    return contested;
+  }
+
+  @Override
+  public void markFirstBlockConfirmed() {
+    firstBlockConfirmed = true;
+  }
+
+  @Override
+  public void markLastBlockConfirmed() {
+    lastBlockConfirmed = true;
+  }
+
+  @Override
+  public void markAsContested() {
+    // TODO: Be more careful about exact behaviour here
```

Comment:
is this going to be resolved in the current issue or are we raising a new issue or just leaving with no issue context?

Reply:
Removed it, the behaviour works well for testing.  I am thinking I may be able to remove `StubBatch` and just use some simple utils with the `SyncSourceBatch` but that will definitely be a later PR.